--- wsgi.py	Fri Sep  5 00:18:56 2008
+++ wsgi.py.new	Fri Sep  5 00:40:57 2008
@@ -76,7 +76,7 @@
     def __init__(self, environ):
         script_name = base.get_script_name(environ)
         path_info = force_unicode(environ.get('PATH_INFO', u'/'))
-        if not path_info or path_info == script_name:
+        if not path_info:
             # Sometimes PATH_INFO exists, but is empty (e.g. accessing
             # the SCRIPT_NAME URL without a trailing slash). We really need to
             # operate as if they'd requested '/'. Not amazingly nice to force
@@ -84,7 +84,9 @@
             #
             # (The comparison of path_info to script_name is to work around an
             # apparent bug in flup 1.0.1. Se Django ticket #8490).
-            path_info = u'/'
+    	    path_info = u'/'
+	if path_info == script_name:
+    	    script_name = u''
         self.environ = environ
         self.path_info = path_info
         self.path = '%s%s' % (script_name, path_info)
